scvi/external/cell2location/_base.py

```diff
@@ -0,0 +1,1060 @@
+from datetime import date
+from typing import Optional
+
+import matplotlib
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import pyro
+import torch
+from pyro import poutine
+from pyro.infer import SVI
+from pyro.infer.autoguide import AutoNormal, init_to_mean
+from tqdm.auto import tqdm
+
+from scvi.dataloaders import AnnDataLoader
+from scvi.model._utils import parse_use_gpu_arg
+from scvi.module.base import PyroBaseModuleClass
+from scvi.train import PyroTrainingPlan, Trainer
+
+from .autoguide import AutoGuideList, AutoNormalEncoder
+
+
+class Cell2locationBaseModule(PyroBaseModuleClass):
+    def __init__(
+        self,
+        model,
+        amortised: bool = False,
+        single_encoder: bool = True,
+        encoder_kwargs=None,
+        data_transform="log1p",
```

Comment:
we have a Literal type you can use for a few string options

Reply:
Can you please give me an example?